nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core-api/src/main/java/org/apache/nifi/groups/FlowSynchronizationOptions.java

```diff
@@ -247,6 +254,15 @@ public Builder componentStopTimeoutAction(final ComponentStopTimeoutAction actio
             return this;
         }
 
+        /**
+         * Specifies a callback whose methods will be called when various updates are made to the flow
+         * @param callback the FlowSynchronizationCallback to use
+         * @return the builder
+         */
+        public Builder callback(final FlowSynchronizationCallback callback) {
```

Comment:
As noted above, as this is designed & used, I think this is more of a listener that can be provided, than a callback that should be required. This should be optionally provided.